### DISCUSSION TITLE
Is it possible to get the actual ref of DragOverlay div?

### DISCUSSION THREAD

**Discussion by yf-hk** [clauderic/dnd-kit#232] [Q&A]
Like for example pass a ref object and forwardRef in DragOverlay?

**Comment by clauderic** [comment]
The `DragOverlay` div ref is an implementation detail of the library, so this isn't something that's possible nor planned. 
What's your use-case?

**Reply by yf-hk** [reply]
One of the main reasons is that I need to resize the overlay during the drag process, depending on the mouse position. Currently in all drag events, `active.rect` is not the actual size if I resized it, so I'll have to manually calculate the real size of the overlay using `.getBoundingClientRect`. See #233. There's no otherways to get it work without a sort of hacky workaround.
The other reason is that I put an image inside of the overlay. I found that every time I drag it, it rerenders the image so there is a glitter. So I manually cloned the image from inside of draggable element and attached it to the overlay div. But I'm not quite sure about this one, maybe there's a better way of doing it

**Reply by clauderic** [reply]
That's currently a use-case that isn't well supported. 

A number of changes need to happen in the library:
- Short-term, you could try to implement this with modifiers, but modifiers don't currently affect the collision rect when you modify the width and height of the element; would happily accept a PR to that effect, just haven't had time to prioritize this.
- Long-term, there should be a `ResizeObserver` in `DragOverlay` to respond to `width` / `height` changes to the rendered children. This wouldn't respond to `transform` changes though, since those don't actually cause the element's dimensions to change, so you'd still need to use modifiers if you were using `transform` to update the size

**Reply by yf-hk** [reply]
Sure, thanks for your reply. Actually currently I managed to get it work using a hacky way.

I put a div with a ref inside of the overlay and use `innerDivRef.current.parentElement` to find the real overlay's `ref`. Then I use that ref to calculate the overlay in the dragMove event to get the real size and position of it.

Everything sort of working fine. It just that the code looks a bit weird and it's hard to mantain.
